service/controller/cluster.go

```diff
@@ -104,6 +104,7 @@ func newClusterResourceSets(config ClusterConfig) ([]*controller.ResourceSet, er
 	{
 		c := ClusterResourceSetConfig{
 			CertsSearcher:      config.CertsSearcher,
+			G8sClient:          config.K8sClient.G8sClient(),
```

Comment:
Since you're passing the full K8sClient into this config as well, you could probably drop this line and get the G8sClient from it inside.

Reply:
I kind of like exposing both clients in this case because it isn't necessarily obvious that a K8sClient would contain a G8sClient, and this way it matches the format of all the other constructors. Going to go with this but open to discussion if someone wants

Reply:
I think it's good to be consistent with passing in only the child `G8sClient`/`K8sClient` or the parent `K8sClient`, but no worries if you prefer this way.